### DISCUSSION TITLE
Upcoming v4.0.0

### DISCUSSION THREAD

**Discussion by klaasnicolaas** [klaasnicolaas/home-assistant-glow#222] [Ideas]
## A new version is coming to town! 🎅🏻 

> [!Note]
> It won't be long before the release will be published! 😄 

Currently working on a new version (v4) with a number of breaking changes. There will be a new documentation website, with more options to display information in a logical manner and you can quickly and easily flash your ESP with the correct firmware with ESP web tools.

./Klaas

## Checklist:
- [x] Improve/refactor documentation website
  - [x] Update hardware list url's
  - [X] Update installation method (esp web tools)
  - [X] Move info from readme to docs website
  - [x] Final touch on docs website (local search, sponsor info, etc.)
- [x] Change of default GPIO pins
  - [x] ESP32 -> GPIO26
  - [x] ESP8266 -> GPIO13 (and other boards)
- [x] Adjust settings more in the UI / Home Assistant entities
  - [x] The pulse rate
- [x] Set the total energy entity equal to the value of your electricity meter
- [X] Seperate YAML file for each board type
- [X] Impove the install via ESP web tools
  - [X] Different install for earch board type / chip family (esp8266, esp32)
  - [X] Use the manifest.json file for esp web tools

**Comment by Liniik** [comment]
Could you please create a new branch for the progress on this version and revert master branch before changes? Usually the master is working code in somehow finished phase and new things or updates have separate branches... Or at least put a message in readme on which tag/branch is the latest working/stable version. Currently I'm confused because the files on master don't reflect state of docs etc. 🤔

**Reply by klaasnicolaas** [reply]
It's not an option for me at this stage to roll back things or work on separate branches. I understand very well how branches are working, but due to certain choices in the past and simplifying the workflow, I've carefully considered the choice to work this way to refactor the project and create a fresh start.

The stable version from `home_assistant_glow.yaml` can be find here: https://github.com/klaasnicolaas/home-assistant-glow/blob/eda2baa1c96b033dc2262be7ec70a49ea7d9dfaf/home_assistant_glow.yaml

I had already announced this on Twitter and if desired I can put a message in the readme.

**Reply by Liniik** [reply]
Ök, I understand the circumstances.

Thanks for the link.

It'd be great if you could put a message in the readme for us, who come directly from the Home Assistant page about energy monitoring to this repo and don't monitor any socials of HA. Thanks!

**Comment by klaasnicolaas** [comment]
Release has been published, time to close this discussion
